disperser/dataapi/metrics.go

```diff
@@ -98,38 +102,97 @@ func (g *Metrics) Start(ctx context.Context) {
 }
 
 type DynamoDBCollector struct {
-	blobMetadataStore *blobstore.BlobMetadataStore
-	blobStatusMetric  *prometheus.Desc
-	logger            common.Logger
+	blobMetadataStore    *blobstore.BlobMetadataStore
+	blobStatusMetric     *prometheus.Desc
+	scrapeDurationMetric *prometheus.GaugeVec
+	logger               common.Logger
 }
 
 func NewDynamoDBCollector(blobMetadataStore *blobstore.BlobMetadataStore, logger common.Logger) *DynamoDBCollector {
-	return &DynamoDBCollector{
+	if blobMetadataStore == nil {
+		logger.Error("BlobMetadataStore is nil, metrics will not be collected")
+	}
+
+	collector := &DynamoDBCollector{
 		blobMetadataStore: blobMetadataStore,
 		blobStatusMetric: prometheus.NewDesc("dynamodb_blob_metadata_status_count",
 			"Number of blobs with specific status in DynamoDB",
 			[]string{"status"},
 			nil,
 		),
+		scrapeDurationMetric: prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
This is monitoring this collector itself, what do you want to use this for?

Reply:
Just want to measure average scrape duration over time.